util/design/lib/OtpMemImg.py

```diff
@@ -0,0 +1,384 @@
+        else:
+            if len(part['items']) == 0:
+                log.warning("Partition does not contain any items.")
+
+        # Check if partition name exists in memory map and get index
+        part_idx = self.get_part_idx(part['name'])
+        if part_idx < 0:
+            raise RuntimeError('Partition {} does not exist'.format(
+                part['name']))
+
+        # Only partitions with a hardware digest can be locked.
+        if part['lock'] and not \
+           self.config['partitions'][part_idx]['hw_digest']:
+            raise RuntimeError(
+                'Partition {} does not contain a hardware digest'.format(
+                    part['name']))
+
+        # Augment memory map datastructure with lock bit.
+        self.config['partitions'][part_idx]['lock'] = part['lock']
+
+    def merge_item_data(self, part, item):
+        '''This validates and merges the item data into the memory map dict'''
+        item.setdefault('name', 'unknown_name')
+        item.setdefault('value', '0x0')
+
+        # Check if item name exists within partition and get index
+        part_idx = self.get_part_idx(part['name'])
+        item_idx = self.get_item_index(part['name'], item['name'])
+        if item_idx < 0:
+            raise RuntimeError('Item {} does not exist'.format(item['name']))
+
+        item_size = self.config['partitions'][part_idx]['items'][item_idx][
+            'size']
+        random_or_hexvalue(item, 'value', item_size * 8)
+        self.config['partitions'][part_idx]['items'][item_idx]['value'] = item[
+            'value']
+
+        log.info('> Adding item {} with size {}B and value:'.format(
+            item['name'], item_size))
+        fmt_str = '{:0' + str(item_size * 2) + 'x}'
+        value_str = fmt_str.format(item['value'])
+        bytes_per_line = 8
+        j = 0
+        while value_str:
+            # Print out max 64bit per line
+            line_str = ''
+            for k in range(bytes_per_line):
+                num_chars = min(len(value_str), 2)
+                line_str += value_str[-num_chars:]
+                if k < bytes_per_line - 1:
+                    line_str += ' '
+                value_str = value_str[:len(value_str) - num_chars]
+            log.info('  {:06x}: '.format(j) + line_str)
+            j += bytes_per_line
+
+    def override_data(self, img_config):
+        '''Override specific partition items'''
+
+        if 'partitions' not in img_config:
+            raise RuntimeError('Missing partitions key in configuration.')
+
+        for part in img_config['partitions']:
+            self.merge_part_data(part)
+            log.info('Overriding values of {} partition.'.format(part['name']))
+            for item in part['items']:
+                self.merge_item_data(part, item)
+
+    def streamout_partition(self, part_idx):
+        '''Scramble and stream out partition data as a list of bytes'''
+
+        part_name = self.config['partitions'][part_idx]['name']
```

Comment:
I'd consider defining `partition = self.config['partitions'][part_idx]` (or indeed taking it as a parameter) to avoid all the square brackets below.

Reply:
that make sense. I pass it in as an arg now (the iteration in the streamout_hexfile function becomes simpler as well)